src/js_lexer.zig

```diff
@@ -836,6 +836,10 @@ fn NewLexer_(
         }
 
         inline fn nextCodepoint(it: *LexerType) CodePoint {
+            if (it.current >= it.source.contents.len) {
+                it.end = it.current;
+                return -1;
+            }
```

Comment:
Did valgrind auto fix this? Very cool!

Reply:
Valgrind found it, the fix is mine (and I messed up at first, forgetting the "it.end = it.current;" line). More reason to vet those changes thoroughly.

Reply:
Which scenario does this happen? We allocate extra padding bytes in the source contents buffer to avoid this extra check in an extremely hot code path.

Reply:
> Which scenario does this happen? We allocate extra padding bytes in the source contents buffer to avoid this extra check in an extremely hot code path.

Where does this happen? It's a good idea, both here and in the toml code below.